### PR TITLE
[#162] Refactor and optimize component rendering

### PR DESCRIPTION
**Pull request by morcosan**
[#162] Split ai-chat store into multiple stores
[#162] Add global log() function in /utils

### CODE REVIEW

File: src-app/biz-modules/ai-chat/_page-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		allSubchatsLoading,
+		chatLoading,
+		subchatLoading,
+		loadActiveSubchat,
+		resetActiveSubchat,
+	} = useAiChat()
+	const [searchParams] = useSearchParams()
+	const navigate = useNavigate()
+
+	const subchatId = parseInt(String(searchParams.get('subchat')))
```
Comment (costinEEST) [inline src-app/biz-modules/ai-chat/_page-panel.tsx:22]: `String()` is not needed, I guess.

File: src-app/biz-modules/ai-chat/_page-layout.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { AppLayout } from '@app/layouts/app-layout'
+import { useUiViewport } from '@ds/release'
+import { ReactNode, useEffect } from 'react'
```
Comment (costinEEST) [inline src-app/biz-modules/ai-chat/_page-layout.tsx:3]: `.. { type ReactNode, ..`

**Review by costinEEST — APPROVED** [review]

File: src-app/biz-modules/ai-chat/_page-panel.tsx (hunk 1 above)
Comment (morcosan) [reply src-app/biz-modules/ai-chat/_page-panel.tsx:22]: I think I got warning from typescript without it. So either string() or type casting.

File: src-app/biz-modules/ai-chat/_page-layout.tsx (hunk 2 above)
Comment (morcosan) [reply src-app/biz-modules/ai-chat/_page-layout.tsx:3]: What do you mean?

**Pull request merged**